experimental/table/server/under/hive/src/main/java/alluxio/table/under/hive/util/PathTranslator.java

```diff
@@ -65,16 +58,21 @@ public PathTranslator addMapping(String alluxioPath, String ufsPath) {
    * @return the corresponding alluxio path
    * @throws IOException if the ufs path is not mounted
    */
-  public String toAlluxioPath(String ufsPath) throws IOException {
-    for (BiMap.Entry<String, String> entry : mPathMap.entrySet()) {
-      if (ufsPath.startsWith(entry.getValue())) {
-        String alluxioPath = entry.getKey() + ufsPath.substring(entry.getValue().length());
-        if (alluxioPath.startsWith("/")) {
-          // scheme/authority are missing, so prefix with the scheme and authority
-          alluxioPath =
-              ConfigurationUtils.getSchemeAuthority(ServerConfiguration.global()) + alluxioPath;
+  public AlluxioURI toAlluxioPath(AlluxioURI ufsPath) throws IOException {
+    for (Map.Entry<AlluxioURI, AlluxioURI> entry : mPathMap.entrySet()) {
+      try {
+        if (entry.getValue().isAncestorOf(ufsPath)) {
+          String alluxioPath = entry.getKey().getPath()
+              + ufsPath.getPath().substring(entry.getValue().getPath().length());
+          if (alluxioPath.startsWith("/")) {
+            // scheme/authority are missing, so prefix with the scheme and authority
+            alluxioPath =
+                ConfigurationUtils.getSchemeAuthority(ServerConfiguration.global()) + alluxioPath;
+          }
+          return new AlluxioURI(alluxioPath);
```

Comment:
Every `AlluxioURI` will parse the string as a URI. Is this something we want to do?